### PR TITLE
Gh420 multipart part size enforcement

### PR DESCRIPTION
**Pull request by slfritchie**
For GH issue #420 

Add multipart size enforcement.

Also, Refactoring: …
- Rewrite enforce_part_size(), thanks to Brett Hazen
- Remove debugging cruft in old enforce_part_size() to throw
  artificial error for parts=5 case
- Wrap QuickCheck stuff with -ifdef(EQC).
- Use a wrapper func to convert throw -> atom false.

Also, add checks for too-small parts and mp upload result = expectation

### CODE REVIEW

File: src/riak_cs_mp_utils.erl [hunk 1]
```diff
@@ -262,7 +267,7 @@ upload_part_finished(Bucket, Key, UploadId, _PartNumber, PartUUID, MD5,
                      Caller, RiakcPidUnW) ->
     Extra = {PartUUID, MD5},
     do_part_common(upload_part_finished, Bucket, Key, UploadId,
-    Caller, [{upload_part_finished, Extra}], RiakcPidUnW).
+                   Caller, [{upload_part_finished, Extra}], RiakcPidUnW).
```
Comment (kuenishi) [inline src/riak_cs_mp_utils.erl:270]: This indentation looks strange. Typo?

**Comment by kellymclaughlin** [comment]
For maximum riak_test happiness when merging these changes with master, can you add this small change as part of this pr:

```
diff --git a/riak_test/src/rtcs.erl b/riak_test/src/rtcs.erl
index b84f8f6..8a4b5f5 100644
--- a/riak_test/src/rtcs.erl
+++ b/riak_test/src/rtcs.erl
@@ -106,7 +106,8 @@ cs_config() ->
        {proxy_get, enabled},
        {anonymous_user_creation, true},
        {riak_pb_port, 10017},
-       {stanchion_port, 9095}
+       {stanchion_port, 9095},
+       {cs_version, 010300}
```

File: src/riak_cs_mp_utils.erl (hunk 1 above)
Comment (slfritchie) [inline src/riak_cs_mp_utils.erl:270]: The old line was indented too far to the left.  The result is:

```
upload_part_finished(Bucket, Key, UploadId, _PartNumber, PartUUID, MD5,
                     Caller, RiakcPidUnW) ->
    Extra = {PartUUID, MD5},
    do_part_common(upload_part_finished, Bucket, Key, UploadId,
                   Caller, [{upload_part_finished, Extra}], RiakcPidUnW).
```

**Comment by kellymclaughlin** [comment]
Looks good to me, +1 to merge

**Pull request merged**